src/librustc_codegen_ssa/mir/rvalue.rs

```diff
@@ -774,12 +774,17 @@ fn cast_float_to_int<'a, 'tcx, Bx: BuilderMethods<'a, 'tcx>>(
     float_ty: Bx::Type,
     int_ty: Bx::Type,
 ) -> Bx::Value {
-    let fptosui_result = if signed { bx.fptosi(x, int_ty) } else { bx.fptoui(x, int_ty) };
-
     if let Some(false) = bx.cx().sess().opts.debugging_opts.saturating_float_casts {
-        return fptosui_result;
+        return if signed { bx.fptosi(x, int_ty) } else { bx.fptoui(x, int_ty) };
+    }
+
+    let try_sat_result = if signed { bx.fptosi_sat(x, int_ty) } else { bx.fptoui_sat(x, int_ty) };
+    if let Some(try_sat_result) = try_sat_result {
+        return try_sat_result;
     }
 
+    let fptosui_result = if signed { bx.fptosi(x, int_ty) } else { bx.fptoui(x, int_ty) };
```

Comment:
While not necessarily immediately clear how to fix this, it'd be nice to avoid the duplication of `fptosui_result`. Not a huge issue though.

Reply:
Should I maybe create a function that deduplicates the if at least? Otherwise I think it either has to be this duplication or unnecessary creation of the non-saturating instruction if the saturating one is going to be used.

Reply:
Eh I wouldn't bend over too much for this, it seems fine either way.